### PR TITLE
enhancement(kubernetes_logs source): Adhere to instrumentation spec

### PR DESCRIPTION
**Pull request by neuronull**
Closes: #14148.
Epic: #13995.

### CODE REVIEW

**Comment by spencergilbert** [comment]
I've been chatting with @neuronull on Slack about this PR and I'm not confident that we actually drop events here - but I'm not confident that we aren't either. I can try and make some time to go back through the source and get myself back up to speed, but otherwise I don't feel like I can give a good review on this change.

**Comment by neuronull** [comment]
> I've been chatting with @neuronull on Slack about this PR and I'm not confident that we actually drop events here - but I'm not confident that we aren't either. I can try and make some time to go back through the source and get myself back up to speed, but otherwise I don't feel like I can give a good review on this change.

:+1: I'll also come back to review this logic some more. The `kubernetes_logs` source won't be :heavy_check_mark: by this PR alone anyway- it still needs the `assert_source_compliance` test helper change. I'm basically tabling this until I get stuck on something.

**Comment by spencergilbert** [comment]
Added labels to run the e2e tests in CI 👍

File: src/sources/kubernetes_logs/mod.rs [hunk 1]
```diff
@@ -525,6 +525,7 @@ impl Source {
                     Err(error) => emit!(KubernetesLifecycleError {
```
Comment (neuronull) [inline src/sources/kubernetes_logs/mod.rs:525]: I'm just assuming here that this is a non-recoverable failure that results in dropped events. I don't actually know for sure.

**Review by spencergilbert — APPROVED** [review]
This seems right, but is definitely complicated. We should try and prioritize this for improved tests or refactoring to improve the observability of the process.

**Pull request merged**